### PR TITLE
Declare the versions of tools allowing for updates by dependabot

### LINKED ISSUE #2284: Declare the versions of tools allowing for updates by dependabot
**What would you like to be cleaned**:

Declare the versions of the tools in `Makefile-deps` so that they are managed by dependabot ([example of hard-coded dependency](https://github.com/kubernetes-sigs/kueue/blob/ceaa3b1fff1e2d95a2d4695d16f99b0d9a8d4ec0/Makefile-deps.mk#L34)).

**Why is this needed**:

To reduce manual maintenance and make sure the tools are not lagging behind. This ensures Kueue is compatible with the latest versions of the tools.

### PR DESCRIPTION
**Pull request by mszadkow**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Move the declaration of version of the tools from make to go.mod

#### Which issue(s) this PR fixes:

Fixes #2284

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: hack/internal/tools/pinversion.go [hunk 1]
```diff
@@ -23,4 +23,19 @@ import (
 	// we need to reference individual dependencies used by it
 	_ "sigs.k8s.io/controller-tools/pkg/crd"
 	_ "sigs.k8s.io/controller-tools/pkg/genall/help/pretty"
+
```
Comment (mbobrovskyi) [inline hack/internal/tools/pinversion.go:26]: Can we remove unnecessary space?

**Comment by mbobrovskyi** [comment]
Overall LGTM

/assign @trasc

File: Makefile-deps.mk [hunk 2]
```diff
@@ -56,22 +65,22 @@ ginkgo: ## Download ginkgo locally if necessary.
...
+	@GOBIN=$(PROJECT_DIR)/bin GO111MODULE=on $(GO_CMD) install github.com/mikefarah/yq/v4@$(YQ_VERSION)
 
 HELM = $(PROJECT_DIR)/bin/helm
 .PHONY: helm
 helm: ## Download helm locally if necessary.
-	@GOBIN=$(PROJECT_DIR)/bin GO111MODULE=on $(GO_CMD) install helm.sh/helm/v3/cmd/helm@v3.12.1
+	@GOBIN=$(PROJECT_DIR)/bin GO111MODULE=on $(GO_CMD) install helm.sh/helm/v3/cmd/helm@$(HELM_VERSION)
 
 GENREF = $(PROJECT_DIR)/bin/genref
 .PHONY: genref
```
Comment (tenzen-y) [inline Makefile-deps.mk:86]: Couldn't we manage the genref as well?

Comment (trasc) [reply Makefile-deps.mk:86]: `genref` is a main package it cannot be added in `go.mod`

Comment (mszadkow) [reply Makefile-deps.mk:86]: Exactly it's `module/command` and not a `package` or something that can be imported :( https://pkg.go.dev/github.com/kubernetes-sigs/reference-docs/genref

**Review by trasc — COMMENTED** [review]
/lgtm

**Comment by alculquicondor** [comment]
/approve

We can split into a separate gomodule in a follow up.

**Pull request merged**